SmartDeviceLink/SDLLifecycleManager.m

```diff
@@ -246,6 +250,51 @@ - (void)didEnterStateConnected {
 }
 
 - (void)didEnterStateRegistered {
+    NSArray<SDLLanguage> *supportedLanguages = self.configuration.lifecycleConfig.languagesSupported;
+    SDLLanguage desiredLanguage = self.configuration.lifecycleConfig.language;
+    SDLLanguage actualLanguage = self.registerResponse.language;
+    BOOL delegateCanUpdateLifecycle = [self.delegate respondsToSelector:@selector(managerWillUpdateLifecycleToLanguage:)];
+    
+    // language mismatch? but actual language is a supported language? and delegate has implemented method?
+    if (actualLanguage != desiredLanguage && [supportedLanguages containsObject:actualLanguage] && delegateCanUpdateLifecycle) {
+        [self.lifecycleStateMachine transitionToState:SDLLifecycleStateUpdatingConfiguration];
+    } else {
+        [self.lifecycleStateMachine transitionToState:SDLLifecycleStateSettingUpManagers];
+    }
+}
+
+- (void)didEnterStateUpdatingConfiguration {
+    // we can expect that the delegate has implemented the update method and the actual language is a supported language
+    SDLLanguage actualLanguage = self.registerResponse.language;
+
+    SDLLifecycleConfigurationUpdate *configUpdate = [self.delegate managerWillUpdateLifecycleToLanguage:actualLanguage];
+    
+    if (configUpdate) {
```

Comment:
Call me old-fashioned but I use [single-entry single-exit](https://en.wikipedia.org/wiki/Single-entry_single-exit) whenever possible an appropriate (except improve legibility and control flow depth).

Return early would duplicate the line to transit to the next state.

Reply:
Understood, but I think it's worth it to [fail fast](https://en.wikipedia.org/wiki/Fail-fast) and keep the [happy path](https://en.wikipedia.org/wiki/Happy_path) as far to the left and stay away from rightward drift if possible. I think this would be a case where `guard` would be used in swift and a return early, so I think we should do the same with obj-c by checking and keeping the happy path to the left for readability's sake.